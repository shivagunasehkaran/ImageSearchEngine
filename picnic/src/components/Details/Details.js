import React from 'react';
import {Text, View} from 'react-native';
import {styles} from '@components/Details/Details.style';
import FastImage from '@components/FastImage';

const Details = props => {
  const randomData = props.randomData;
  let URL = null;
  let rating = null;
  if (randomData && randomData.images && randomData.images.downsized_large) {
    URL = randomData.images.downsized_large.url;
  }
  if (randomData) {
    rating = randomData.rating;
  }
  return (
    <View style={styles.container}>
      <View style={styles.fastImage}>
        <FastImage uri={URL} />
      </View>
      <View>
        <Text style={styles.textContent}>
          {randomData ? randomData.title : ''}
        </Text>
        <Text style={styles.textContent}>{URL}</Text>
        {rating && (
          <View style={styles.rating}>
            <Text style={styles.ratingText}>{rating}</Text>
          </View>
        )}
      </View>
    </View>
  );
};

export default Details;
