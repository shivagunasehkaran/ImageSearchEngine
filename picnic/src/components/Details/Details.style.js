import {ColourPalette} from '@assets/styles/ColoursStyles';
import {StyleSheet} from 'react-native';
import {moderateScale} from 'react-native-size-matters';

export const styles = StyleSheet.create({
  container: {
    backgroundColor: ColourPalette.white,
  },
  fastImage: {
    margin: 30,
    height: moderateScale(300),
  },
  textContent: {
    margin: 15,
    fontSize: 15,
  },
  rating: {
    width: moderateScale(60),
    height: moderateScale(60),
    borderRadius: moderateScale(60) / 2,
    backgroundColor: ColourPalette.gray,
    justifyContent: 'center',
    alignItems: 'center',
    left: 300,
  },
  ratingText: {
    fontSize: 25,
    color: ColourPalette.white,
  },
});
