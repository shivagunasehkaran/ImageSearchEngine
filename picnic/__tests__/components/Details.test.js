import Details from '@components/Details/Details';
import enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

enzyme.configure({adapter: new Adapter()});

describe('Image details component validation', () => {
  it('renders correctly with image detail elments', async () => {
    const wrapper = shallow(<Details />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with text elements', async () => {
    const wrapper = shallow(<Details />);
    expect(wrapper.find('Text')).toHaveLength(3);
  });

});
