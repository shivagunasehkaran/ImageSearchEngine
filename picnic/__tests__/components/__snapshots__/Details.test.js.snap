// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image details component validation renders correctly with image detail elments 1`] = `ShallowWrapper {}`;
